File: Problems/Algorithms/1356. Sort Integers by The Number of 1 Bits/sort_by_number_1_bits.py
class Solution:
    def sortByBits(self, arr: List[int]) -> List[int]:
        arr.sort(key=lambda x: (self.number_of_1(x), x))
        return arr

    def number_of_1(self, x: int) -> int:
        ans = 0

        while x > 0:
            ans += 1
            x &= x-1

        return ans

File: Problems/Algorithms/1356. Sort Integers by The Number of 1 Bits/SortByNumber1Bits.kt
class Solution {
    fun sortByBits(arr: IntArray): IntArray {
        return arr.toList().sortedWith(compareBy({ numberBits(it) }, { it })).toIntArray()
    }
    
    private fun numberBits(x: Int): Int {
        var ans = 0
        
        var curr = x
        while (curr > 0) {
            ans++
            curr = curr and (curr-1)
        }
        
        return ans
    }
}

